### PR TITLE
future warning for with

### LINKED ISSUE #245: add deprecation warning when argument passed to `--with` contains hyphens (`-`)
in #164 we no longer allow hyphens passed to `--with`, instead, users should use underscores.

to give them time to update their code, we should throw a FutureWarning whenever they pass an argument that contains hyphens and tell them to use underscores instead

### PR DESCRIPTION
**Pull request by Palashio**
## Describe your changes

Adds deprecation warning if hypens are passed to the with parameter. 

## Issue ticket number and link
Closes #245 

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added thorough [tests](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#testing) (when necessary).
- [ ] I have added the right documentation in the [docstring](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#documenting-changes-and-new-features) and [changelog](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--249.org.readthedocs.build/en/249/

### CODE REVIEW

File: src/sql/store.py [hunk 1]
```diff
@@ -81,6 +81,12 @@ def __init__(self, store: SQLStore, query: str, with_: Iterable = None):
         self._query = query
         self._with_ = with_ or []
 
+        if any(map(lambda x: "-" in x, self._with_)):
+            raise FutureWarning(
```
Comment (edublancas) [inline src/sql/store.py:85]: warnings are displayed using warnings.warn: https://docs.python.org/3/library/warnings.html - here you're raising an error

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/sql/store.py (hunk 1 above)
Comment (Palashio) [reply src/sql/store.py]: Just made this modification.

**Comment by Palashio** [comment]
it seems like it's erroring on readthedocs.org?

File: src/sql/store.py [hunk 2]
```diff
@@ -81,6 +82,13 @@ def __init__(self, store: SQLStore, query: str, with_: Iterable = None):
         self._query = query
         self._with_ = with_ or []
 
+        if any(map(lambda x: "-" in x, self._with_)):
+            raise warnings.warn(
```
Comment (Mause) [inline src/sql/store.py:86]:
```suggestion
            warnings.warn(
```

File: src/sql/store.py [hunk 3]
```diff
@@ -81,6 +82,13 @@ def __init__(self, store: SQLStore, query: str, with_: Iterable = None):
         self._query = query
         self._with_ = with_ or []
 
+        if any(map(lambda x: "-" in x, self._with_)):
```
Comment (Mause) [inline src/sql/store.py:85]:
```suggestion
        if any("-" in x for x in self._with_):
```

**Comment by edublancas** [comment]
> it seems like it's erroring on readthedocs.org?

yes, cause you sill have a `raise` statement. should be removed.

also, let's display in the error message the proposed solution. e.g., if they have `--with no-nulls`, let's suggest them switching it to `--with no_nulls`

**Review by edublancas — CHANGES_REQUESTED** [review]
looks like it isn't working?

image

please add a unit test

**Comment by edublancas** [comment]
oh, I realized I said we needed it for `--with`; but we also need it for `--save`

**Comment by Palashio** [comment]
> oh, I realized I said we needed it for `--with`; but we also need it for `--save`

just added this to the `--save` argument.

**Review by edublancas — APPROVED** [review]

**Pull request merged**